webapp/channels/src/components/user_settings/notifications/user_settings_notifications.tsx

```diff
@@ -223,45 +230,42 @@ export default class NotificationsTab extends React.PureComponent<Props, State>
         data.comments = this.state.notifyCommentsLevel;
         data.auto_responder_active = this.state.autoResponderActive.toString() as UserNotifyProps['auto_responder_active'];
         data.auto_responder_message = this.state.autoResponderMessage;
+        data.first_name = this.state.firstNameKey.toString() as UserNotifyProps['first_name'];
+        data.channel = this.state.channelKey.toString() as UserNotifyProps['channel'];
 
         if (!data.auto_responder_message || data.auto_responder_message === '') {
-            data.auto_responder_message = localizeMessage(
-                'user.settings.notifications.autoResponderDefault',
-                'Hello, I am out of office and unable to respond to messages.',
-            );
+            data.auto_responder_message = this.props.intl.formatMessage({
+                id: 'user.settings.notifications.autoResponderDefault',
+                defaultMessage: 'Hello, I am out of office and unable to respond to messages.',
+            });
         }
 
-        const mentionKeys = [];
+        const mentionKeys: string[] = [];
         if (this.state.usernameKey) {
             mentionKeys.push(this.props.user.username);
         }
-
-        let stringKeys = mentionKeys.join(',');
-        if (this.state.customKeys.length > 0 && this.state.customKeysChecked) {
-            stringKeys += ',' + this.state.customKeys;
+        if (this.state.isCustomKeysWithNotificationInputChecked && this.state.customKeysWithNotification.length > 0) {
+            this.state.customKeysWithNotification.forEach((key) => {
+                mentionKeys.push(key.value);
+            });
         }
-
-        data.mention_keys = stringKeys;
-        data.first_name = this.state.firstNameKey.toString() as UserNotifyProps['first_name'];
-        data.channel = this.state.channelKey.toString() as UserNotifyProps['channel'];
+        data.mention_keys = mentionKeys.join(',');
 
         this.setState({isSaving: true});
-        NotificationSounds.stopTryNotificationRing();
-
-        this.props.actions.updateMe({notify_props: data} as UserProfile).
-            then(({data: result, error: err}) => {
-                if (result) {
-                    this.handleUpdateSection('');
-                    this.setState(getNotificationsStateFromProps(this.props));
-                } else if (err) {
-                    this.setState({serverError: err.message, isSaving: false});
-                }
-            });
+        stopTryNotificationRing();
+
+        const {data: updatedUser, error} = await this.props.updateMe({notify_props: data}) as ActionResult<Partial<UserProfile>, ServerError>;
```

Comment:
Why the casting is needed?

Reply:
this will be fixed in MM-46907, in very future :P